DCHdigi/test/test_DCHdigi/test_DCHdigi.sh

```diff
@@ -0,0 +1,26 @@
+# file: test_DCHdigi.sh
+# author: Alvaro Tolosa-Delgado, CERN 2024
+# to run: sh + test_DCHdigi.sh
+# goal: run sim-digitizer of the DCH v2, and return code printed by check_DCHdigi_output.py
+
+# run simulation with the drift chamber alone
+ddsim --steeringFile sim_steering.py --outputFile 'dch_proton_10GeV.root' -N 10 --runType batch --random.seed 42
+
+# download file for cluster counting technique
+ifilename="https://fccsw.web.cern.ch/fccsw/filesForSimDigiReco/IDEA/DataAlgFORGEANT.root"
+wget $ifilename
+
+# Check if the input file exists
+if [[ ! -f "$(basename $ifilename)" ]]; then
+    echo "Error: File '$(basename $ifilename)' not found."
+    exit 1
+fi
+
+# run digitizer for position smearing and cluster counting calculation
+k4run runDCHdigi.py
+
+# check distribution of distance from hit position to the wire
+check_DCHdigi_output=$( (python3 check_DCHdigi_output.py) 2>&1)
+
+# return value printed out by the previous python script
+exit $check_DCHdigi_output
```

Comment:
I am sorry, I do not fully understand what it means 'run with source'. After a quick google search, it might be caused by missing shebang, just added it, lets see if that fixes the problem

Reply:
I mean that when one does `source test_DCHdigi.sh`, the ssh connection closes due to the call to `exit`? If you really need this exit code, run it in a subshell to avoid closing the session

```suggestion
(exit $check_DCHdigi_output)
```

Reply:
If you need the exit code to propagate to the calling shell, you can also simply use the fact, that the return code of the full script will be the return code of the last command. Hence, you can also simply run `python3 check_DCHdigi_output.py > 2>&1` as last command and it will have the same effect.

Reply:
many thanks, done!